pkg/exclusions/exclusions_test.go

```diff
@@ -0,0 +1,303 @@
+package exclusions
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/osv-scanner/internal/sbom"
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+type MockReporter struct {
+	Output string
+	Error  string
+}
+
+func (r *MockReporter) PrintText(text string) {
+	r.Output = text
+}
+func (r *MockReporter) PrintError(text string) {
+	r.Error = text
+}
+
+func (r *MockReporter) HasPrintedError() bool {
+	return false
+}
+func (r *MockReporter) PrintResult(vulnResult *models.VulnerabilityResults) error {
+	return nil
+}
+
+func TestExcludePackages(t *testing.T) {
+	t.Parallel()
+	type testCase struct {
+		name                 string
+		lockfileData         []lockfile.PackageDetails
+		exclusionPatterns    []string
+		expectedLockfileData []lockfile.PackageDetails
+		wantErr              bool
+	}
+	tests := []testCase{
+		{
+			name:                 "Package matches regex pattern",
+			lockfileData:         []lockfile.PackageDetails{{Name: "package1"}, {Name: "package2"}, {Name: "package3"}},
+			exclusionPatterns:    []string{"package1"},
+			expectedLockfileData: []lockfile.PackageDetails{{Name: "package2"}, {Name: "package3"}},
+			wantErr:              false,
+		},
+		{
+			name:                 "Multiple packages matches regex pattern",
+			lockfileData:         []lockfile.PackageDetails{{Name: "package1"}, {Name: "package2"}, {Name: "somethingDifferent"}},
+			exclusionPatterns:    []string{".*package.*"},
+			expectedLockfileData: []lockfile.PackageDetails{{Name: "somethingDifferent"}},
+			wantErr:              false,
+		},
+		{
+			name:                 "No packages match regex pattern",
+			lockfileData:         []lockfile.PackageDetails{{Name: "package1"}, {Name: "package2"}, {Name: "package3"}},
+			exclusionPatterns:    []string{"package4"},
+			expectedLockfileData: []lockfile.PackageDetails{{Name: "package1"}, {Name: "package2"}, {Name: "package3"}},
+			wantErr:              false,
+		},
+		{
+			name:                 "Invalid regex pattern",
+			lockfileData:         []lockfile.PackageDetails{{Name: "package1"}, {Name: "package2"}, {Name: "package3"}},
+			exclusionPatterns:    []string{"*hello*"},
+			expectedLockfileData: []lockfile.PackageDetails{{Name: "package1"}, {Name: "package2"}, {Name: "package3"}},
+			wantErr:              true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			mockReporter := &MockReporter{}
+			lockfile := lockfile.Lockfile{Packages: tt.lockfileData}
+			filteredLockfile, err := ExcludePackages(mockReporter, tt.exclusionPatterns, &lockfile)
+
+			if !cmp.Equal(err != nil, tt.wantErr) {
+				t.Errorf("ExcludePackages returned an unexpected error: %v", err)
+			}
+
+			for i, pkg := range filteredLockfile.Packages {
```

Comment:
You might be able to do a direct compare of the array. No need to iterate over it.